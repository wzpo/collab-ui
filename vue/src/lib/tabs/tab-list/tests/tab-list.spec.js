import { mount } from '@vue/test-utils';
import TabList from '../index.vue';
import Tab from '../../tab/index.vue';
import Vue from 'vue';

Vue.component(Tab.name, Tab);

describe('TabList', () => {
  it('should match snapshot', () => {
    const wrapper = mount(TabList, {
      provide: {
        getActiveIndex: () => 0,
        getFocusIndex: () => 0,
        onFocus: () => {},
        onActivate: () => {}
      },
      slots: {
        default: `<md-tab heading="one"/>`
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render children', () => {
    const container = mount(TabList, {
      provide: {
        getActiveIndex: () => 0,
        getFocusIndex: () => 0,
        onFocus: () => {},
        onActivate: () => {}
      },
      slots: {
        default: `<md-tab heading="one"/>`
      }
    });

    expect(container.findAll('.md-tab__item').length).toEqual(1);
  });
});

