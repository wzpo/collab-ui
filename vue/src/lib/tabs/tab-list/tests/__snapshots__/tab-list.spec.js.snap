// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabList should match snapshot 1`] = `
<ul role="tablist" class="md-tab__list">
  <li tabindex="-1" class="md-tab__item active"><a role="tab" href="javascript:void(0)" tabindex="0" aria-current="true">one</a></li>
</ul>
`;
